prow/plugins/project/project.go

```diff
@@ -0,0 +1,270 @@
+		Description: "The project plugin allows members of a configurable GitHub team to set the project and column on an issue or pull request.",
+		Config: func(repos []string) map[string]string {
+			configMap := make(map[string]string)
+			for _, repo := range repos {
+				team, exists := config.ProjectConfig[repo]
+				if exists {
+					configMap[repo] = msgForTeam(team)
+				}
+			}
+			configMap[""] = msgForTeam(config.ProjectConfig[""])
+			return configMap
+		}(enabledRepos),
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/project <board>, /project <board> <column>, or /project clear <board>",
+		Description: "Updates the project board and column for an issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Members of the project maintainers GitHub team can use the '/project' command.",
+		Examples:    []string{"/project 0.5.0", "/project 0.5.0 3999801", "/project clear 0.4.0"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, &e, pc.PluginConfig.ProjectConfigMap)
+}
+
+func buildProjectMap(projects []github.Project) map[string]int {
+	m := make(map[string]int)
+	for _, project := range projects {
+		m[project.Name] = project.ID
+	}
+	return m
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, projectConfigMap map[string]plugins.ProjectConfig) error {
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	matches := projectRegex.FindStringSubmatch(e.Body)
+	if len(matches) != 2 && len(matches) != 3 {
+		return nil
+	}
+
+	proposedProject := matches[1]
+	proposedColumn := -1
+	if len(matches) == 3 && proposedProject != clearKeyword {
+		columnInt, err := strconv.Atoi(matches[2])
+		if err == nil {
+			proposedColumn = columnInt
+		}
+	}
+
+	// See if the current user is part of the team in order to set the project/column
+	projectConfig, exists := projectConfigMap[fmt.Sprintf("%s/%s", org, repo)]
+	if !exists {
+		// fallback default
+		projectConfig = projectConfigMap[""]
+	}
+	projectMaintainers, err := gc.ListTeamMembers(projectConfig.ProjectMaintainersID, github.RoleAll)
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, person := range projectMaintainers {
+		login := github.NormLogin(e.User.Login)
+		if github.NormLogin(person.Login) == login {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// not in the project maintainers team
+		msg := fmt.Sprintf(mustBeSigLead, org, project.ProjectMaintainersTeam, org, project.ProjectMaintainersTeam)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	var projects []Project
+
+	repos, err := gc.GetRepos(org, false)
+	if err != nil {
+		log.WithError(err).Errorf("Error listing all repos in org %s", org)
+		return err
+	}
+	// Get all projects for all repos
```

Comment:
Can I add an issue to a project from a different repo?

Reply:
As long as the project and the issue/PR are under the same org, we can add the issue/PR to a project under a different repo https://github.com/kubernetes/test-infra/issues/10514#issuecomment-456476170

Reply:
Eeek